Application/Helper/Controller.hs

```diff
@@ -1,5 +1,59 @@
 module Application.Helper.Controller where
 
+import Data.Map qualified as Map
+import Generated.Types
 import IHP.ControllerPrelude
+import Web.Types
 
--- Here you can add functions which are available in all your controllers
\ No newline at end of file
+-- | an `Exercise` with all associated MuscleGroups
+fetchExerciseWithMuscleGroups :: (?modelContext :: ModelContext) => Id Exercise -> IO ExerciseWithMuscleGroups
+fetchExerciseWithMuscleGroups exerciseId = do
+  exercise <- fetch exerciseId
+
+  muscleGroups <-
+    query @MuscleGroup
+      |> innerJoin @ExercisesMuscleGroup (#id, #muscleGroupId)
+      |> filterWhereJoinedTable @ExercisesMuscleGroup (#exerciseId, exerciseId)
+      |> fetch
+
+  return $
```

Comment:
Just a quick note that idiomatic IHP code uses `pure` instead of `return` (as `return` might be confusing to e.g. JS developers)